src/charts/time_series_labeler.coffee

```diff
@@ -56,112 +85,156 @@ Ember.Charts.TimeSeriesLabeler = Ember.Mixin.create
+  # Dynamic x labelling tries to intelligently limit the number of labels
+  # along the x axis to a specified limit. In order to do this, time
+  # specificity is callibrated (e.g. for a range of 2 years, instead of having
+  # the labels be in years, the specificity is increased to quarters)
+  # If the minTimeSpecificity or maxTimeSpecificity are set, then the labels
+  # are limited to fall between the time units between these bounds.
+  dynamicXLabelling: (start,stop) ->
+    ind1 = @get('DOMAIN_ORDERING').indexOf(@get 'minTimeSpecificity')
+    ind2 = @get('DOMAIN_ORDERING').indexOf(@get 'maxTimeSpecificity')
+    
+    # Refers to the metrics used for the labelling
+    domainTypes = @get('DOMAIN_ORDERING')[ind1..ind2]
 
-  weeksBetween: (start, stop, skip = 1) ->
-    d3.time.weeks(start, stop).filter((d, i) -> !(i % skip))
+    # The labeller type to create the labels for each metric
+    labellerTypes =
+      domainTypeToLabellerType[domainType] for domainType in domainTypes
 
-  secondsBetween: (start, stop, skip = 1) ->
-    d3.time.seconds(start, stop).filter((d, i) -> !(i % skip))
+    #The time span in various metrics
+    times = (numTimeBetween[domainTypeToLongDomainType[d]](start, stop) for d in domainTypes)
 
-  # the weeks which should be labelled
-  labelledWeeks: (start, stop) ->
-    weeks = @weeksBetween(start, stop)
+    labels = null
+    xLabels = @get 'maxNumberOfLabels'
 
-    # if we have too many weeks then we reduce to the maximum number of
-    # weeks labelled such that they are uniformly distributed in the
-    # range
-    if weeks.length > @get('maxNumberOfLabels')
-      skipVal = Math.ceil(weeks.length / @get('maxNumberOfLabels'))
-      @weeksBetween(start, stop, skipVal)
-    else
-      weeks
+    # a ratio specifying the point at which the time specificity should be
+    # increased. The specificity ratio roughly bounds the number of x axis
+    # labels:
+    #
+    # [SPECIFICITY_RATIO*(MAX_LABELS) , MAX_LABELS]
+    #
+    # Essentially, the higher the ratio (until a max of 1), the closer the
+    # number of labels along the x axis is to the max number of labels
+    SPECIFICITY_RATIO = 0.7
 
-  labelledDays: (start, stop) ->
-    days = d3.time.days(start, stop)
-
-    if days.length > @get 'maxNumberOfLabels'
-      skipVal = Math.ceil(days.length / @get('maxNumberOfLabels'))
-      d3.time.days(start, stop).filter((d, i) -> !(i % skipVal))
-    else
-      days
-
-  labelledHours: (start, stop) ->
-    hours = d3.time.hours(start, stop)
-
-    if hours.length > @get 'maxNumberOfLabels'
-      skipVal = Math.ceil(hours.length / @get('maxNumberOfLabels'))
-      d3.time.hours(start, stop).filter((d, i) -> !(i % skipVal))
-    else
-      hours
+    for timeBetween, i in times
+      interval = null
+      if timeBetween < xLabels
+        # quarter labels are calculated by simply getting month labels with 3
+        # month gaps
+        interval = if domainTypes[i] is 'Q' then 3 else 1
```

Comment:
should we change this number 3 into a constant? The constant's name will be meaningful and self-explained.